include/eve/detail/function/simd/common/gather_high.hpp

```diff
@@ -0,0 +1,45 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2020 Joel FALCOU
+  Copyright 2020 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/pattern.hpp>
+#include <eve/forward.hpp>
+
+namespace eve::detail
+{
+  template<simd_value Wide> EVE_FORCEINLINE auto gather_high_(EVE_SUPPORTS(cpu_), Wide v) noexcept
+  {
+    // Well, we dunno how to do this, so we just reshuffle
+    return v[gather_high_n<cardinal_v<Wide>>];
```

Comment:
? It will end up here again or not? How is this helpful?

Reply:
No, you won't end there. The [] will catch the proper gather_low call, the one below